addrmgr/addrmanager.go

```diff
@@ -900,86 +999,95 @@ func (a *AddrManager) Good(addr *wire.NetAddress) {
 	ka.lastattempt = now
 	ka.attempts = 0
 
-	// move to tried set, optionally evicting other addresses if needed.
+	// If the address is already tried then return since it's already good.
+	// Otherwise, move it to a tried bucket. If the target tried bucket is full,
+	// then room will be made by evicting the oldest address in that bucket and
+	// moving it to a new bucket. If the psuedorandomly selected new bucket is
+	// full, then swap the addresses' positions between tried and new.
 	if ka.tried {
-		return
+		return nil
 	}
 
-	// ok, need to move it to tried.
-
 	// remove from all new buckets.
 	// record one of the buckets in question and call it the `first'
 	addrKey := NetAddressKey(addr)
-	oldBucket := -1
+	availableNewBucketIndex := -1
```

Comment:
`availableNewBucketIndex` is a confusing variable name for me because it does not refer to `addrNew` explicitly, calling it a new bucket instead. I do not have a good variable name at the moment. Something along the lines of `addrNewOpenSlot` or `addrNewAvailableIndex`   would be better I think.

Reply:
Updated